source/m2mtlvserializer.cpp

```diff
@@ -164,9 +177,44 @@ bool M2MTLVSerializer::serialize_multiple_resource(const M2MResource *resource,
     return success;
 }
 
+/* See, OMA-TS-LightweightM2M-V1_0-20170208-A, Appendix C,
+ * Data Types, Integer, TLV Format 
+ * 
+ * This is easy to change to support 8, 16 or 32 interger. */
+template <typename T> void M2MTLVSerializer::set_value_int(T value, uint8_t *buffer, uint32_t &size)
+{
+    size = sizeof(value);
+
+    /* write bytes to big endian order in buffer */
+    for (uint8_t ind=0;ind<size;ind++) {
+        buffer[size-ind-1] = ((value >> (ind*8)) & 0xFF);
+    }
+}
+
+template void M2MTLVSerializer::set_value_int<int8_t>(int8_t value, uint8_t *buffer, uint32_t &size);
```

Comment:
I agree that if you're not fixed size, those functions may not be for you anyway.

But those simple utility functions should never be stubbed for unit tests, they should be pulled in to the build and working, just like you have working pointer accesses, linked lists and memcpy.

Doing anything else would be insane.

What you stub should be a matter of common sense, not the totally arbitrary rule "is it in libc or not".

@JanneKiiskila - this keeps coming up, due to bonkers definition of "unit test", which is apparently "It must compile only 1 C file, and use no libraries, but libc is permitted".

I once saw a unit tests for packet parsing where all the 8-bit values are read from the packet, as 8-bit accesses are not stubbed, but the 16-bit values were not because common_read_16_bit() was stubbed.  If anyone with a test hat is still trying to force this nonsense, they need a firm kick.